extensions/amp-dailymotion/1.0/component.js

```diff
@@ -0,0 +1,112 @@
+import * as Preact from '#preact';
+import {forwardRef} from '#preact/compat';
+import {useMemo} from '#preact';
+import {dispatchCustomEvent} from '#core/dom';
+import {
+  DailymotionEvents,
+  getDailymotionIframeSrc,
+  makeDailymotionMessage,
+} from '../dailymotion-api';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+import {parseQueryString} from '#core/types/string/url';
+
+/**
+ * @param {string} method
+ * @return {string}
+ */
+function makeMethodMessage(method) {
+  switch (method) {
+    case 'mute':
+      return makeDailymotionMessage('muted', [true]);
+    case 'unmute':
+      return makeDailymotionMessage('muted', [false]);
+    case 'showControls':
+      return makeDailymotionMessage('controls', [true]);
+    case 'hideControls':
+      return makeDailymotionMessage('controls', [false]);
+    default:
+      return makeDailymotionMessage(method);
+  }
+}
+
+/**
+ * @param {MessageEvent} event
+ */
+function onMessage({currentTarget, data}) {
+  const parsedData = parseQueryString(/** @type {string} */ (data));
+  const event = parsedData?.['event'];
+  if (event === DailymotionEvents.PAUSE) {
+    dispatchCustomEvent(currentTarget, 'pause');
+  } else if (event === DailymotionEvents.PLAY) {
+    dispatchCustomEvent(currentTarget, 'playing');
+  } else if (event === DailymotionEvents.END) {
+    dispatchCustomEvent(currentTarget, 'pause');
+    dispatchCustomEvent(currentTarget, 'end');
+  } else if (event === DailymotionEvents.API_READY) {
+    dispatchCustomEvent(currentTarget, 'canplay');
+  }
```

Comment:
FYI you can use `redispatch` here 
```suggestion
redispatch(currentTarget, event, {
    [DailymotionEvents.API_READY]: 'canplay',
    [DailymotionEvents.END]: [VideoEvents.ENDED, VideoEvents.PAUSE],
    [DailymotionEvents.PAUSE]: VideoEvents.PAUSE,
    [DailymotionEvents.PLAY]: VideoEvents.PLAYING,
  });
```

Reply:
It seems like`redispatch` only works in an amp environment but i've update the `dispatchCustomEvent` calls to be individual calls instead of sending the array.